File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringBuilder result = new StringBuilder();

        int n = Integer.parseInt(br.readLine().trim());
        int[] nums = new int[n];

        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(br.readLine().trim());
        }

        Stack<Integer> stack = new Stack<>();
        int current = 1;
        boolean isYes = true;

        for (int num : nums) {
            while (current <= num) {
                stack.add(current);
                result.append('+').append('\n');
                current++;
            }
            if(stack.peek() == num){
                stack.pop();
                result.append('-').append('\n');
            }else{
                isYes = false;
                break;
            }
        }

        if(isYes) bw.write(String.valueOf(result));
        else bw.write("NO\n");
        bw.flush();
        br.close();
        bw.close();
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

input = sys.stdin.readline

n = int(input().strip())

result = list()
nums = [int(input().strip()) for _ in range(n)] 

stack = []
result = []
current = 1
isYes = True

for num in nums:
    while current <= num:
        stack.append(current)
        result.append('+')
        current += 1
    if stack[-1] == num:
        stack.pop()
        result.append('-')
    else:
        isYes = False
        break

if isYes:
    sys.stdout.write('\n'.join(map(str, result)) + '\n')
else:
    sys.stdout.write('NO' + '\n')